.github/actions/restore-node/check-git-status.sh

```diff
@@ -1,11 +1,26 @@
 #!/bin/bash
+
 # Set verbose execution
 set -x
+
 # Navigate to the specified directory
 cd "$1" || exit $?
+
+# Get the value of IGNORE_ENDO_BRANCH
+ignore_dirty_yarn_lock=$2 || exit $?
+
 # Check for unexpected changes
 # Fail if git status detects changes
 changes=$(git status --porcelain)
+
+if [ "$ignore_dirty_yarn_lock" = true ]; then
```

Comment:
My bash is rusty, but how does this work?

Reply:
`[ a = b ]` returns successful (exit code 0) if the two strings 'a' and `b` are identical. That works because `/bin/[` is an executable program, as well as Bourne-like shells having `[' as a built-in command.

Reply:
Ah yeah so `true` in this case is the string `"true"` ? That's the part that confused me, the whole quoting thing is optional since it's just arguments

Reply:
I use quoting as a safety net. If the variable is empty or unset, quoting prevents syntax errors. For example:

```bash
ignore_dirty_yarn_lock=""
if [ $ignore_dirty_yarn_lock = true ]; then
  echo "True"
fi
```

This will result in an error: `line 6: [: =: unary operator expected`.

Reply:
> Ah yeah so true in this case is the string "true" ?

Yes. I think Bash supports boolean expressions for comparisons and conditions. But it primarily allows you to declare and compare strings and numbers.